### PR TITLE
Add torch `RMSProp` optimizer

### PR DESCRIPTION
**Pull request by natolambert**
# What does this PR do?

Add torch `RMSProp` for easy use in TRL library, particularly to match the default use case of Direct Preference Optimization.
Script is [here](https://github.com/huggingface/trl/blob/main/examples/dpo.py).

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@younesbelkada and I discussed this.

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
This looks great to me! Thanks a lot for this @natolambert

**Review by ArthurZucker — APPROVED** [review]
Good to me as well, just wondering why we did not have it before? cc @pacman100

**Comment by natolambert** [comment]
Thanks, I can't merge it as I don't have write access, so @ArthurZucker or @younesbelkada can do so!

**Pull request merged**
